rag_agent.py

```diff
@@ -1,66 +1,61 @@
 # Uses Llama3.1 model from Ollama
 
 import os
-import warnings, logging
-
+import warnings
 from langchain_community.vectorstores import FAISS
-from langchain_community.embeddings import HuggingFaceEmbeddings, OpenAIEmbeddings
+from langchain_community.embeddings import HuggingFaceEmbeddings
 from langchain_community.document_loaders import PyMuPDFLoader, TextLoader
-from langchain.chains import RetrievalQA
-from langchain_core.prompts import ChatPromptTemplate
+from langchain_core.runnables import RunnablePassthrough
+from langchain_core.output_parsers import StrOutputParser
 from langchain_ollama.llms import OllamaLLM
-
+from langchain.prompts import ChatPromptTemplate
 
 # Suppress future warnings related to deprecations
 warnings.filterwarnings("ignore", category=FutureWarning)
 warnings.filterwarnings("ignore", category=DeprecationWarning)
 
+# Document paths
 document_paths = [
-    './docs/Pygame Documentation.pdf'),
-    './docs/Python GTK+3 Documentation.pdf'),
+    '/home/kshitij/Downloads/Sugarlabs/Pippy-Activity/Pygame Documentation.pdf',
+    '/home/kshitij/Downloads/AI-model/Python GTK+3 Documentation.pdf',
 ]
 
-PROMPT = """
-You are a highly intelligent Python coding assistant with
-access to both general knowledge and specific Pygame documentation.
-1. You only have to answer Python and GTK based coding queries.
-2. Prioritize answers based on the documentation when the query is related to it.
-   However make sure you are not biased towards documentation provided to you.
-3. Make sure that you don't mention words like context or documentation
-   stating what has been provided to you.
-4. Provide step-by-step explanations wherever applicable.
-5. If the documentation does not contain relevant information, use your
-   general knowledge.
-6. Always be clear, concise, and provide examples where necessary.
-"""
-
-TEMPLATE = f"""{PROMPT}
-Question: {{question}}
+# Revised Prompt Template to avoid mentioning the source
+PROMPT_TEMPLATE = """
+You are a highly intelligent Python coding assistant. 
+You are ONLY allowed to answer Python and GTK-based coding questions. 
+1. Focus on coding-related problems, errors, and explanations.
+2. Use the knowledge from the provided Pygame and GTK documentation without explicitly mentioning the documents as the source.
+3. Provide step-by-step explanations wherever applicable.
+4. If the documentation does not contain relevant information, use your general knowledge.
+5. Always be clear, concise, and provide examples where necessary.
+
+Question: {question}
 Answer: Let's think step by step.
 """
 
-
-class RAG_Agent():
+class RAG_Agent:
     def __init__(self):
         self.model = None
         self.retriever = None
-        self.prompt = ChatPromptTemplate.from_template(TEMPLATE)
+        self.prompt = ChatPromptTemplate.from_template(PROMPT_TEMPLATE)
 
-    def set_model(self, model):
+    def set_model(self, model="llama3.1"):
+        """Set the Llama 3.1 model from Ollama."""
```

Comment:
My comment above makes this redundant, you can set a docstring that's better as you now understand why the function exists.